color/quoted.c

```diff
@@ -122,3 +122,402 @@ bool quoted_colors_parse_color(enum ColorId color, uint32_t fg, uint32_t bg,
   *rc = MUTT_CMD_SUCCESS;
   return true;
 }
+
+/**
+ * qclass_free - Free a single QClass
+ * @param ptr QClass to free
+ */
+static void qclass_free(struct QClass **ptr)
+{
+  if (!ptr || !*ptr)
+    return;
+
+  struct QClass *qc = *ptr;
+
+  FREE(&qc->prefix);
+  FREE(ptr);
+}
+
+/**
+ * qclass_free_tree - Free a tree of QClass
+ * @param[out] quote_list Quote list to free
+ */
+void qclass_free_tree(struct QClass **quote_list)
+{
+  struct QClass *next = NULL;
+
+  while (*quote_list)
+  {
+    if ((*quote_list)->down)
+      qclass_free_tree(&((*quote_list)->down));
+    next = (*quote_list)->next;
+    qclass_free(quote_list);
+    *quote_list = next;
+  }
+}
+
+/**
+ * qclass_new - Create a new QClass
+ * @retval ptr New QClass
+ */
+static struct QClass *qclass_new(void)
+{
+  struct QClass *qc = mutt_mem_calloc(1, sizeof(struct QClass));
+
+  return qc;
```

Comment:
As there's no complicated initialization logic and the function providers type details, any objection to `return mutt_mem_calloc(...)`?

Reply:
done